ssi-sd-jwt/src/encode.rs

```diff
@@ -0,0 +1,218 @@
+use jose_b64::base64ct::{Base64UrlUnpadded, Encoding};
+use rand::{CryptoRng, Rng};
+use serde::Serialize;
+use ssi_jwk::{Algorithm, JWK};
+
+use crate::*;
+
+fn encode_disclosure_with_salt<ClaimValue: Serialize>(
+    salt: &str,
+    claim_name: Option<&str>,
+    claim_value: &ClaimValue,
+) -> Result<String, serde_json::Error> {
+    let disclosure = match claim_name {
+        Some(claim_name) => serde_json::json!([salt, claim_name, claim_value]),
+        None => serde_json::json!([salt, claim_value]),
```

Comment:
I think the claim_name is mandatory for a disclosure right?

Reply:
No, array style disclosures don't require a name.